### ISSUE TITLE
PCA and ICA return ValueError

### ISSUE THREAD

**Issue by bainzo** [thunder-project/thunder-factorization#4]
Running a simple test,

``` Python
import thunder as td
from factorization import PCA,ICA
ts = td.series.fromexample('fish')
algo = PCA()
w,t = algo.fit(ts)
```

fails with,

``` Python-traceback
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-350246bb024c> in <module>()
----> 1 w,t = algopca.fit(ts)

/home/ec2-user/anaconda2/lib/python2.7/site-packages/factorization/base.pyc in fit(self, X)
     15 
     16         if data.mode == "local":
---> 17             results = self._fit_local(data)
     18 
     19         if data.mode == "spark":

/home/ec2-user/anaconda2/lib/python2.7/site-packages/factorization/algorithms/PCA.pyc in _fit_local(self, X)
     14 
     15     def _fit_local(self, X):
---> 16         t, v = self._fit_spark(X)
     17         return t.toarray(), v.toarray()
     18 

/home/ec2-user/anaconda2/lib/python2.7/site-packages/factorization/algorithms/PCA.pyc in _fit_spark(self, X)
     23         from thunder.series import fromarray
     24 
---> 25         X = toseries(X).center(1)
     26 
     27         svd = SVD(k=self.k, method=self.svd_method, max_iter=self.max_iter, tol=self.tol, seed=self.seed)

/home/ec2-user/anaconda2/lib/python2.7/site-packages/factorization/utils.pyc in toseries(y)
     11 
     12     if len(y.shape) != 2:
---> 13         raise ValueError("factorization on for 2-dimensional arrays")
     14 
     15     return y

ValueError: factorization on for 2-dimensional arrays
```

In both PySpark and regular python.

I've also had similar errors for ICA with this type of test and my own tif data.

I'm running Thunder 1.0.0 and Spark 1.6.0.

**Comment by bainzo** [comment]
I've also had this issue when running with my own tif data.

**Comment by jwittenbach** [comment]
@bainzo This is one of the aspects that we want to address before we publish the package. With the original design, all of the factorization algorithms only work with two-dimensional objects. It's easy enough of the user to ravel/unravel higher-dimensional data to on passing it into/out of the algorithms, but we realized that we should probably just do that for them automatically.

**Comment by bainzo** [comment]
@jwittenbach Ahh, that makes sense.

Is there a method like `flatten` that can reshape the Series object?

**Comment by bainzo** [comment]
I've opened this as [issue #335 on the main thunder repo](https://github.com/thunder-project/thunder/issues/335) as suggested by @jwittenbach

**Comment by jwittenbach** [comment]
Just merged a big update that address this (#5). Now factorization can be done direction on `Images` and `Series` objects from `thunder` and shapes will be conserved. I'm going to close this, but definitely open a new issue if there are any problems with the new functionality~
